python/cudf/groupby/groupby.py

```diff
@@ -113,7 +113,7 @@ def __init__(self, df, by, method="hash", as_index=True, level=None):
                 if self._df.index.names is None or sum(
                         x is None for x in self._df.index.names) > 1:
                     self._df_index_names = list(
-                            range(len(self._df_index.levels)))
+                            range(len(self._df.index.levels)))
```

Comment:
Do we need to materialize the levels here? Could this be `len(self._df.index._source_data.columns)`?

Reply:
Good idea